jsapp/js/components/permissions/userPermissionRow.es6

```diff
@@ -100,14 +102,25 @@ class UserPermissionRow extends React.Component {
             });
           }
 
+          let permName = '???';
+          // TODO simplify this code when https://github.com/kobotoolbox/kpi/issues/2332 is done
+          if (this.props.kind === ASSET_KINDS.get('asset')) {
+            if (this.props.assignablePerms.has(perm.permission)) {
+              permName = this.props.assignablePerms.get(perm.permission);
+            }
+          }
+          if (this.props.kind === ASSET_KINDS.get('collection')) {
+            permName = COLLECTION_PERMISSIONS[permConfig.getPermission(perm.permission).codename];
+          }
+
           return <bem.UserRow__perm
             title={perm.description}
-            key={perm.name}
+            key={permName}
           >
-            {perm.name}
+            {permName}
 
             {permUsers.length > 0 &&
-              ' (' + permUsers.join(', ') + ')'
+              String.fromCharCode(160) + '(' + permUsers.join(', ') + ')'
```

Comment:
What about add it to `KEY_CODES' constant as well?

Reply:
Good thinking!